engines/twp/lighting.cpp

```diff
@@ -61,7 +61,8 @@ void main(void) {
 	vec2 curPixelPosInLocalSpace = vec2(pixelPos.x, -pixelPos.y);
 
 	vec3 diffuse = vec3(0, 0, 0);
-	for (int i = 0; i < u_numberLights; i++) {
+	for (int i = 0; i < 50; i++) {
+		if(i >= u_numberLights){break;}
```

Comment:
```suggestion
		if (i >= u_numberLights)
			break;
```

Reply:
This (and the rest of the checks) could also be written like this:

    for (int i = 0; i < 50 && i < u_numberLights; i++) {

Reply:
I thought that would be nice as well (and you'd think that this would allow it to statically unroll the loop), but unfortunately the loop condition needs to take the form of `loop_index relational_operator constant_expression` (Appendix A, 4 Control Flow). Using multiple conditions throws the following error:

> 'for' : Invalid condition